src/main/java/com/stripe/net/ApiRequestParams.java

```diff
@@ -70,6 +80,116 @@ public EnumParam read(JsonReader in) {
     }
   }
 
+  /**
+   * Type Adapter to handle deserializing extra params. The main goal is to flatten field
+   * that contains extra params.
+   */
+  private static class HasExtraParamsTypeAdapterFactory implements TypeAdapterFactory {
+    public Map<String, Object> deserializeJsonObject(JsonWriter jsonWriter,
+                                                     JsonObject jsonObject) throws IOException {
+      Map<String, Object> objMap = new HashMap<>();
+      jsonWriter.beginObject();
+      for (Map.Entry<String, JsonElement> entry : jsonObject.entrySet()) {
+        String key = entry.getKey();
+        jsonWriter.name(key);
+        JsonElement element = entry.getValue();
+        writeJsonElement(jsonWriter, element);
+      }
+      jsonWriter.endObject();
+      return objMap;
+    }
+
+    private void writeJsonElement(JsonWriter jsonWriter, JsonElement element) throws IOException {
+      if (element.isJsonNull()) {
+        // this is needed because the json delegated at this point would only have null
+        // due to empty enum
+        boolean oldSerializedNull = jsonWriter.getSerializeNulls();
+        jsonWriter.setSerializeNulls(true);
+        jsonWriter.nullValue();
+        jsonWriter.setSerializeNulls(oldSerializedNull);
+      } else if (element.isJsonObject()) {
```

Comment:
handling null as a result of the other type adapter converting empty param to null